server/chat-plugins/friends.ts

```diff
@@ -0,0 +1,911 @@
+/**
+ * Friends list plugin.
+ * Allows for adding and removing friends, as well as seeing their activity.
+ * Written by Mia.
+ * @author mia-pi-git
+ */
+import type {Database, Transaction, Statement} from 'better-sqlite3';
+import {Utils} from '../../lib/utils';
+import {FS} from '../../lib/fs';
+import {QueryProcessManager} from '../../lib/process-manager';
+import {Repl} from '../../lib/repl';
+import {Config} from '../config-loader';
+
+/** Max friends per user */
+const MAX_FRIENDS = 100;
+/** Max friend requests. */
+const MAX_REQUESTS = 6;
+const REQUEST_EXPIRY_TIME = 30 * 24 * 60 * 60 * 1000;
+const MAX_PROCESSES = 1;
+
+interface DatabaseRequest {
+	statement: string;
+	type: 'all' | 'get' | 'run' | 'transaction';
+	data: AnyObject | any[];
+}
+
+interface DatabaseResult {
+	/** if there's an error, this is it. */
+	error?: string;
+	result?: any;
+}
+
+const STATUS_COLORS: {[k: string]: string} = {
+	idle: '#ff7000',
+	online: '#009900',
+	busy: '#cc3838',
+};
+
+const STATUS_TITLES: {[k: string]: string} = {
+	online: 'Online',
+	idle: 'Idle',
+	busy: 'Busy',
+	offline: 'Offline',
+};
+
+export const Friends = new class {
+	/** `Map<oldID, newID> */
+	renames: Map<string, string>;
```

Comment:
Appears to be a redundant cache for the database `friend_renames` table, my previous comment applies